src/interfaces/User.interface.ts

```diff
@@ -28,26 +28,26 @@ export interface User {
   _id?: mongoose.Types.ObjectId;
   firebaseUid: string;
   notificationToken?: string;
-  imgUrlProfileSmall?: string;
-  imgUrlProfileLarge?: string;
-  imgUrlCover?: string;
+  imgUrlProfileSmall: string;
+  imgUrlProfileLarge: string;
+  imgUrlCover: string;
   firstName: string;
   lastName: string;
   email: string;
   phone: string;
-  isActiveNow?: boolean;
-  isEmergency?: boolean;
-  birthday?: string;
+  isActiveNow: boolean;
+  isEmergency: boolean;
+  birthday: string;
   currentGroup?: mongoose.Types.ObjectId;
-  invitedGroups?: mongoose.Types.ObjectId[];
-  friends?: mongoose.Types.ObjectId[];
-  friendRequests?: mongoose.Types.ObjectId[];
-  sentFriendRequests?: mongoose.Types.ObjectId[];
+  invitedGroups: mongoose.Types.ObjectId[];
+  friends: mongoose.Types.ObjectId[];
+  friendRequests: mongoose.Types.ObjectId[];
+  sentFriendRequests: mongoose.Types.ObjectId[];
   lastActive?: LastActive;
-  savedGroups?: SavedGroup[];
-  emergencyContacts?: EmergencyContact[];
-  sentPings?: mongoose.Types.ObjectId[];
-  receivedPings?: mongoose.Types.ObjectId[];
+  savedGroups: SavedGroup[];
+  emergencyContacts: EmergencyContact[];
+  sentPings: mongoose.Types.ObjectId[];
+  receivedPings: mongoose.Types.ObjectId[];
```

Comment:
any reason why we are requiring these fields now? what happens if a document is created without them?

Reply:
We don't do that now :) We will use empty arrays to reduce the amount of undefined checks so now we can treat them as arrays.

Reply:
Best explanation is that if you have an array with one object id and you remove it, it will become an empty array, not undefined. So then an empty array and undefined represent the same thing. Would rather just stick to one.